kite-tools-parent/kite-tools/src/main/java/org/kitesdk/tools/TransformTask.java

```diff
@@ -96,63 +96,32 @@ public TransformTask setNumWriters(int numWriters) {
   }
 
   public PipelineResult run() throws IOException {
-    boolean runInParallel = true;
-    if (isLocal(from.getDataset()) || isLocal(to.getDataset())) {
-      runInParallel = false;
-    }
-
     PType<T> toPType = ptype(to);
     MapFn<T, T> validate = new CheckEntityClass<T>(to.getType());
 
-    if (runInParallel) {
-      TaskUtil.configure(getConf())
-          .addJarPathForClass(HiveConf.class);
-
-      Pipeline pipeline = new MRPipeline(getClass(), getConf());
-
-      PCollection<T> collection = pipeline.read(CrunchDatasets.asSource(from))
-          .parallelDo(transform, toPType).parallelDo(validate, toPType);
-
-      if (compact) {
-        // the transform must be run before partitioning
-        collection = CrunchDatasets.partition(collection, to, numWriters);
-      }
-
-      pipeline.write(collection, CrunchDatasets.asTarget(to), Target.WriteMode.APPEND);
-
-      PipelineResult result = pipeline.done();
+    TaskUtil.configure(getConf())
+        .addJarPathForClass(HiveConf.class);
 
-      StageResult sr = Iterables.getFirst(result.getStageResults(), null);
-      if (sr != null && MAP_INPUT_RECORDS != null) {
-        this.count = sr.getCounterValue(MAP_INPUT_RECORDS);
-      }
-
-      return result;
-
-    } else {
-      Pipeline pipeline = MemPipeline.getInstance();
```

Comment:
Nit: It looks like when we remove the `MemPipeline` code, quite a few imports can be removed at the same time.